### DISCUSSION TITLE
Search text across all sessions in current OS window

### DISCUSSION THREAD

**Discussion by pr3c0g** [kovidgoyal/kitty#5152] [Q&A] [answered]
Hey there!

I've been going through the issues and discussions, and kitty documentation, trying to find a solution for me, but I haven't been able to create one.

I would like to search for text across all currently open kitty terminals, or in case that is not possible, all kitty windows/tabs on the current OS window (I tend to stick to one anyways).

The purpose is finding the right ssh session for example, so after searching I just need to focus that particular window. 

Is this something I could achieve with the scrollback buffer or kitty shell in some way?

I hope this isn't a useless question that has been addressed in the documentation and that I missed it.

Thank you in advance!

**Comment by kovidgoyal** [comment, marked as answer]
You can use remote control to do this. 
https://sw.kovidgoyal.net/kitty/remote-control/ 
in particular kitty @ get-text

or you can write your own kitten for it
https://sw.kovidgoyal.net/kitty/kittens/custom/ 

On Tue, May 31, 2022 at 08:39:43AM -0700, pr3c0g wrote:
> Hey there!
> 
> I've been going through the issues and discussions, and kitty documentation, trying to find a solution for me, but I haven't been able to create one.
> 
> I would like to search for text across all currently open kitty terminals, or in case that is not possible, all kitty windows/tabs on the current OS window (I tend to stick to one anyways).
> 
> The purpose is finding the right ssh session for example, so after searching I just need to focus that particular window. 
> 
> Is this something I could achieve with the scrollback buffer or kitty shell in some way?
> 
> I hope this isn't a useless question that has been addressed in the documentation and that I missed it.
> 
> Thank you in advance!
> 
> -- 
> Reply to this email directly or view it on GitHub:
> https://github.com/kovidgoyal/kitty/discussions/5152
> You are receiving this because you are subscribed to this thread.
> 
> Message ID: ***@***.***>

-- 
_____________________________________

Dr. Kovid Goyal 
https://www.kovidgoyal.net
https://calibre-ebook.com
_____________________________________

**Reply by pr3c0g** [reply]
I did miss that in the documentation. Thank you very much for taking the time to reply.
